control/autoware_vehicle_cmd_gate/src/vehicle_cmd_gate.cpp

```diff
@@ -457,10 +459,21 @@ void VehicleCmdGate::onTimer()
     }
   }
 
+  const auto CheckTopicContinuity = []<class T>(const T & prev_topic, const T & current_topic) {
+    return (rclcpp::Time(current_topic.stamp) - rclcpp::Time(prev_topic.stamp)).seconds() > 0.0
+             ? current_topic
+             : prev_topic;
+  };
+
+  prev_commands_.turn_indicator =
+    CheckTopicContinuity(prev_commands_.turn_indicator, turn_indicator);
+  prev_commands_.hazard_light = CheckTopicContinuity(prev_commands_.hazard_light, hazard_light);
+  prev_commands_.gear = CheckTopicContinuity(prev_commands_.gear, gear);
```

Comment:
The default value of stamp is set to 0, so it will likely work as expected. However, evaluating prev_commands when it is unassigned is not ideal. How about using a shared_ptr?

ex.
https://github.com/autowarefoundation/autoware.universe/blob/main/localization/twist2accel/include/twist2accel/twist2accel.hpp#L53
https://github.com/autowarefoundation/autoware.universe/blob/d4f365da2f2e6403381ce5c64e60d3820428128a/localization/twist2accel/src/twist2accel.cpp#L75

Reply:
The current code uses the default value of turn_indicator, hazard_light and gear of Commands for each mode as long as no topic is received, so I would consider updating pre_command when a topic is received.

Reply:
I have changed the method of checking the continuity. Please review this !
https://github.com/autowarefoundation/autoware.universe/pull/8084/commits/6ced54f7e7ce6d2097f588ac214b6a6b5c8fe0e9